feature/selector.py

```diff
@@ -270,6 +273,102 @@ def _validate(self):
                                                        CatBoostClassifier, CatBoostRegressor)),
                            ValueError("Unknown tree-based estimator" + str(self.estimator)))
 
+    class TextBased(NamedTuple):
+        """
+        # TODO  review the content to double check everything is %100 correct once ALL the implementation is done
+        Text-based method for selecting features/columns from (X_textual, Y).
+        Unlike other selection method, this method operates on textual data, X_textual.
+        In this context, each column/feature in X_textual can be referred to an item with textual description.
+        Similarly, Y represents the labels that are covered by each item.
+        In the test_text.py, X_textual defines as data and Y defines as labels.
+
+        The paper [1] provides a multi-objective optimization problem that is based on
+        set covering formulation. The idea is to select the minimum number of items from X_textual that
+        maximizes the item diversity of the text featurization of items (controlled by the featurization method)
+        while also maximizing the coverage of labels given in Y.
+        If no Y is given, it selects #TODO..
+
+        References:
+        [1] Kadioglu et. al., Optimized Item Selection to Boost Exploration for Recommender Systems, CPAIOR'21
+
+        The parameters corresponding to the methods in the paper [1]
+        random:
+            trails, seed, verbose
+            * random(t=max_num_features): num_features
+            * random(t=unicost()): num_features=None, cost_metric
+            * random(t=diverse()): num_features=None, featurization_method=TextWiser, cost_metric
+        kmeans:
+        greedy:
+        unicost:
+        diverse:
+        max_cover:
+
+        Randomness:
+        Behavior for #TODO methods is non-deterministic, depends on seed.
+
+        Attributes
+        ----------
+        num_features : Num, optional
+            * If num_feature is integer value, select max_num_features.
+            * If num_feature is None, num_features defines by solving a set cover problem using unicost or diverse
+            The algorithm might choose a less number of features depending on other parameters.
+
+        featurization_method :
+            TextWiser object to featurize items in X_textual (it is named data in test and this method does not
+            have impact on the following options:
+                - Random optimization with num_features = max_num_features
```

Comment:
The Pydoc documentation is currently not up-to-date for all optimization methods. However, I will update it for each method as soon as I create a pull request for it.

Regarding the "max_num_features" parameter, I have made the decision to change it to "num_features". This is because this parameter can now be set to any number less than or equal to the maximum number of features that the user is able to select.